### ISSUE TITLE
[wasm] Enforce HarfBuzz cluster guarantees on shaping output

### ISSUE THREAD

**Issue by khaledhosny** [harfbuzz/harfbuzz#4789]
HarfBuzz has some guarantees for the cluster values, especially for levels 0 and 1, but Wasm shaper might not respect these guarantees and HarfBuzz should sanitize the output and enforce them since downstream code that uses HarfBuzz is likely to depend on this.

We might also need to expose the cluster level to the Wasm shaper so it have the chance to work accordingly.

**Comment by behdad** [comment]
Yeah this is a bit unfortunate. The HB guarantees go beyond just monotonicity. The indices are use to index into string memories and out-of-range numbers can do Bad Things™.

**Comment by khaledhosny** [comment]
Yes, I think HarfBuzz should sanitize the clusters. Clip out of range clusters and merge out of order clusters. Or alternative, it should check the output and if it violates the cluster guarantees, it fails the shaping.

**Comment by behdad** [comment]
> it fails the shaping.

It had not occurred to me. Maybe it's best to do that in trivially bad cases, and just fix the monotonity if needed.

Also, maybe if monotone-graphemes is requested, we can do Unicode clustering algorithm as well.
